### DISCUSSION TITLE
Not Work C++ Mxnet and opencv integration

### DISCUSSION THREAD

**Discussion by kotbegemot** [apache/incubator-mxnet#16661] [Q&A] [answered]
How to pass a lot of cv:: Mat to MXDataIter ?
``C++

    auto val_iter = MXDataIter("ImageRecordIter")
            .SetParam("data_shape", Shape(3, image_size, image_size))
            .SetParam("batch_size", batch_size)
            .SetParam("shuffle", 1)
            .SetParam("flat", 1)
            .SetParam("ctx", "cpu")
            .CreateDataIter();

    int count = 0;
    Accuracy acc;
    val_iter.Reset();
    while (val_iter.Next()) {
        auto data_batch = val_iter.GetDataBatch();
        data_batch.data.CopyTo(&args_map["data"]);
        NDArray::WaitAll();
        exec->Forward(false);
        acc.Update(data_batch.label, exec->outputs[0]);
    }

    delete exec;
    MXNotifyShutdown();

    return 0;
``

**Comment by samskalicky** [comment]
@mxnet-label-bot remove [Bug]

**Comment by samskalicky** [comment]
@mxnet-label-bot add [Question]

**Comment by samskalicky** [comment, marked as answer]
@zhreshold any idea?

Heres an example:
https://discuss.gluon.ai/t/topic/7951
